records/labels/spec.go

```diff
@@ -0,0 +1,43 @@
+package labels
+
+import (
+	"fmt"
+)
+
+// LabelSeparator delimits concatenated labels.
+const LabelSeparator = '.'
+
+// Spec is a label specification interface
+type Spec interface {
+	// Mangle transforms an input string to produce an output string that is
+	// compliant with the specification. If the input string is already
+	// compliant then it is returned unchanged.
+	Mangle(string) string
+}
+
+// SpecFunc implements Spec
+type SpecFunc func(string) string
+
+func (sf SpecFunc) Mangle(input string) string {
+	return sf(input)
+}
+
+type HostNameSpec int
+
+const (
+	// see http://www.rfc-base.org/txt/rfc-952.txt
+	HostNameSpec952 HostNameSpec = iota
```

Comment:
FWIW `802.11ac` is IEEE not RFC. RFCs follow a pure monotonic incremental number: http://www.rfc-editor.org/rfc-index2.html